charts/multicluster-controlplane/templates/secret.yaml

```diff
@@ -0,0 +1,38 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: controlplane-config
+  namespace: {{ .Release.Namespace }}
+type: Opaque
+stringData:
+  ocmconfig.yaml: |-
+    apiserver:
+      externalHostname: {{ .Values.apiserver.externalHostname }}
+      port: {{ .Values.apiserver.port }}
+      {{- if and .Values.apiserver.caFile .Values.apiserver.caKeyFile }}
+      caFile: "/controlplane_config/{{ .Values.apiserver.caFile }}"
```

Comment:
the `Values.apiserver.caFile` is a file path or a file name?

we only need the file name here, right?

Reply:
user may have values configuration, like
```
apiserver:
   caFile: "/home/user/certs/myca.crt"
   caKey: "/home/user/certs/myca.key"
```

what is the result of this secret after helm rendered this secret

Reply:
hi, @skeeey  I made a test for this case, here's the result:

the directory 
```
charts/multicluster-controlplane
├── aa
│   └── multicluster_ca
│       ├── ca.crt
│       └── ca.key
├── Chart.yaml
├── templates
│   ├── clusterrolebinding-admin.yaml
│   ├── clusterrolebinding.yaml
│   ├── clusterrole.yaml
│   ├── deployment.yaml
│   ├── pvc.yaml
│   ├── rolebinding.yaml
│   ├── role.yaml
│   ├── route.yaml
│   ├── secret.yaml
│   ├── serviceaccount.yaml
│   └── service.yaml
└── values.yaml
```

values:
```
apiserver:
  caFile: "aa/multicluster_ca/ca.crt"
  caKeyFile: "aa/multicluster_ca/ca.key"
```

the dry-run result:
```
# Source: multicluster-controlplane/templates/secret.yaml
apiVersion: v1
kind: Secret
metadata:
  name: controlplane-config
  namespace: test-helm-3
type: Opaque
stringData:
  ocmconfig.yaml: |-
    apiserver:
      externalHostname: 
      port: 443
      caFile: "/controlplane_config/ca.crt"
      caKeyFile: "/controlplane_config/ca.key"
    etcd:
      mode: embed
      prefix: test-helm-3

data:
  ca.crt: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSURHRENDQWdDZ0F3SUJBZ0lVS2JiYnF5ajYyQWl3Q1NTTFZvMFVvQnMvR0d3d0RRWUpLb1pJaHZjTkFRRUwKQlFBd0pERWlNQ0FHQTFVRUF4TVpiWFZzZEdsamJIVnpkR1Z5TFdOdmJuUnliMnh3YkdGdVpUQWVGdzB5TXpBMApNRGN3T0RVMU1EQmFGdzB5T0RBME1EVXdPRFUxTURCYU1DUXhJakFnQmdOVkJBTVRHVzExYkhScFkyeDFjM1JsCmNpMWpiMjUwY205c2NHeGhibVV3Z2dFaU1BMEdDU3FHU0liM0RRRUJBUVVBQTRJQkR3QXdnZ0VLQW9JQkFRRFEKbGRnWitTUVFBcU04bGFRSy9weWxKRXF5T0J0QThpN05HMkdIcVBlQkVuSk15RXJ1NVhrY2RHQ0R2NVhNaEUrdApvQkJjOW5ZZHRQZkNVRktMTnJRRzNqSXI5RTNFTnJiSG1XcHZUZUprZG91VE1OZ2VDTW5PZWZvaXhXWFE5YXB6CmNyYnM2dVROdlJuK0o2Qk42SktJbVZkUW1JWW5LdzNDbExPRS9pRXJUamNUVW5Cb3EwTDJrai9TKzU1M0JKd0gKNmxoc3dNeFZVZjJTWFZJdkZVYkNVeVkxQWM2eUJhd2UvamVGZW5EU1FDcTh2WTlyQ0ZqV3M4cVJZQ1dPK0hNQQpkWHFLWXNJRm12bC9maHFqWldGWkJjR1B1emVwd0E5cWRodmc2KzFPZkw1c3dhTENtUHpUZlJpTDU2QTczN1RMCk55K2NHRG5HVFJuVXlSUGlia1BoQWdNQkFBR2pRakJBTUE0R0ExVWREd0VCL3dRRUF3SUJCakFQQmdOVkhSTUIKQWY4RUJUQURBUUgvTUIwR0ExVWREZ1FXQkJTZDduT0hJSHhPZlh6Q3pNc2U4eWlCQVJEOFNUQU5CZ2txaGtpRwo5dzBCQVFzRkFBT0NBUUVBdnVtUVVnSHlVb2tiN2duRVAwQkw0QTdXZStiMVZTL3BQa2dSWVJyRVZqZmhvalFZCmJ2cTdvajJ6R0lCcTMrazlwNkVBdys2bFpQejlBRGNLMHR4dnBjTGF5bVQwVkVqMWVmOGRCZmF1SjFDYnV0RUwKeTFFbUVOcXRGdkc0OWdUZCs1K1VndmVYVExWVk1uRit1dlU1bW1WSG9YRjdmdHI0NG5nTnZsTFhSV2x2TmJlMwprOXEyck1BaGZUbGZoYlArMk9NT0hBOFY2YWx4cVY3dFM5RUIyZ1lNcURkaTRRbVZ6Mmd4RFA3cWlIMkMxcE4rCjR6bmRRMTdabDJwSnFyOGozc0hFOUR2NUNxN0ZnUVJsa1RxUG13a1NocG4wTVFWTlRJcXhhMFI5NHZpdDM3SjkKWkNQdUFMenh2am5Tb0ZvWXNESi9kSGMxRGFuOW1CbGhiaVBGK0E9PQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==
  ca.key: LS0tLS1CRUdJTiBSU0EgUFJJVkFURSBLRVktLS0tLQpNSUlFb3dJQkFBS0NBUUVBMEpYWUdma2tFQUtqUEpXa0N2NmNwU1JLc2pnYlFQSXV6UnRoaDZqM2dSSnlUTWhLCjd1VjVISFJnZzcrVnpJUlByYUFRWFBaMkhiVDN3bEJTaXphMEJ0NHlLL1JOeERhMng1bHFiMDNpWkhhTGt6RFkKSGdqSnpubjZJc1ZsMFBXcWMzSzI3T3JremIwWi9pZWdUZWlTaUpsWFVKaUdKeXNOd3BTemhQNGhLMDQzRTFKdwphS3RDOXBJLzB2dWVkd1NjQitwWWJNRE1WVkg5a2wxU0x4Vkd3bE1tTlFIT3NnV3NIdjQzaFhwdzBrQXF2TDJQCmF3aFkxclBLa1dBbGp2aHpBSFY2aW1MQ0JacjVmMzRhbzJWaFdRWEJqN3MzcWNBUGFuWWI0T3Z0VG55K2JNR2kKd3BqODAzMFlpK2VnTzkrMHl6Y3ZuQmc1eGswWjFNa1Q0bTVENFFJREFRQUJBb0lCQURZNjd5OTVqZjIzb2RjcgpsTFhqRXJrQTRZYktmWk42STk3aHdvNnh5SCsvV3daSmdPbnhzWFpMYjVoeFpESURLSEFjMElSSzg5U3ZhMGNTCitSTUd5dkdLTFlNdUJsWUFYNjh2WFBQSlRSWVBtQ1JScTY3TFhDU1pwQVlNU25kMGw3ZGlXTnJxQzBqT1pJYW0KcU0vNHFyaG5IY052bEZsMG9TUTQ0eCt0cnRRQ0RuQnJhQnNiR0dwNmMvQ205ZmJGbk1PKzdlR1A5QmRmd21FYgpzQmc3aTdkWVVHTXRWcE9FajQzTjE5SEc2T2xxRE9VTDZRZXZqcEZWVUlWczVkU2ZwL2QvbmFibHlvVmc5SzVlCnF5b29VZ3lwNUZvZ0JFczdBUGhLZU1hdzlObThZY21tek5DZzBTck95bExnVEdQRnJPNGhBUnNZMyswL1k1M2wKRW1TQWI5RUNnWUVBMTcyd3FLTFc3MnlpUURnVnlLOVlrZmRZWUNVZWFyNmhXQS82VGEyTzdmaFdwNWhRUUJubgpQTDRLNEdqd3k2cDhHQ2xRT205LzA1c0ZPUktHTlV3RlhKMVZQYlQxeFFzSDJ3YWVJTXVVYVVzSkE5bTYrc0ZVCmE0TTJJWDcvdnlGUDFzem5DRkt1V2p2WnVuM3FrZFFpUnRDKzluOVBQU01SUjBoTWc4QmlVSFVDZ1lFQTk0SlIKRTU5L3puV2daS1JzdXNGU3RpSWZqTW5IR3g2V2F1NXRDU29wMzJpc1VFamNudHBNa3VRTUQ1M0VKNWVJWEhMKwpEM2lPb2EwSEJOZG1DV3RqUjh6TWFod0E1WTJrRktRdlU1bmt4Sml4YUtDMHRFWE9zVnhYMWZJczZZVVFmR0ZNCmQwSTdwa2Rvb3Aya1hRNkNZMnRSakFBUzZBVzZobDc3ZzdiWnVEMENnWUJraEo0MXhFRk9wcGZtUWZ0c1dMNTAKckFFQ011azhKUHU1c3lBVlp5TXNIMFFJckQvZnpmSGViRkZYQTY0RnhyWXc3N2JRc3ppa21CclQzNXltWW5neApFYUV3c2dxOHR3bis4OU5qUmxjc3crZzhZZ3JyWUg1NzVWT29RYXJaRzZVK1dKUGdYZDJjYnp4ZjF1RWU5dTdOCmZkOHJyRjk0QmNtOEk2UjYwbjA4TFFLQmdRRFFzbFZpMG02Vnk2RCt1NVhpc3ZmbkUwYS9FN1NMTFhOdm5yTFoKczIrekxUTitUckdZVUVHWTZpa3hVYm9BR0VtbWlsK204K1J3ZHRFZ1IrK0ZxY01BMjdMUkZlTzdxMzdhMk1YLwpIQmhGbk9xYWFrZDBhaGdxZk02alc1eDdHdlFYcklNcFcvT245VFpKUjlzN2kreHRvQ1FXaEVsczBzZkFYWDUrClJVTzNrUUtCZ0d4NnNTLzhOR2VsWmZHSzlkaDVYQUR1V3NGaU52a28yemE0YU1vcmpwL2MrNWpMVHBVWU9MenQKM0JQdmJxcVpTeVQ2TGQvK3FVd3pzQ1IvVSs4VHhCNFFiWkNOVGtsU2duNE9CTkd4N1hEZDlmb3BxK2I5WGJMNwpTOVZsSlZzNEtKY3RGbkN4emFTQlQyZzNzUU9qMEdwU2VLeE9LT28zYWg1Qkk2Nnl3b1psCi0tLS0tRU5EIFJTQSBQUklWQVRFIEtFWS0tLS0tCg==
```

Reply:
@ycyaoxdu would you delete the ca and key content in the comments, thanks

Reply:
done

Reply:
for `aa/multicluster_ca/ca.crt`, the `aa` dir should be in the chart dir, or it can be outside? and can we use the absolute path? like `/home/user/aa/multicluster_ca/ca.crt`

Reply:
OK, you use the `.Values.apiserver.caFile | base` to get file name, I think the absolute path is supported